signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWithAppleButton.kt

```diff
@@ -27,38 +26,34 @@ class SignInWithAppleButton @JvmOverloads constructor(
 
     init {
         val attributes =
-            context.theme.obtainStyledAttributes(attrs, R.styleable.SignInWithAppleButton, 0, 0)
+            context.theme.obtainStyledAttributes(attrs, R.styleable.SignInWithAppleButton, 0, R.style.SignInWithAppleButton)
 
-        val colorStyleIndex =
-            attributes.getInt(
-                R.styleable.SignInWithAppleButton_sign_in_with_apple_button_colorStyle,
-                SignInColorStyle.BLACK.ordinal
-            )
-        val colorStyle = SignInColorStyle.values()[colorStyleIndex]
+        // Style
+        val background = attributes.getDrawable(R.styleable.SignInWithAppleButton_android_background)
+        val icon = attributes.getDrawable(R.styleable.SignInWithAppleButton_android_drawableLeft)
+        val textColor = attributes.getColorStateList(R.styleable.SignInWithAppleButton_android_textColor)
 
+        // Text type
         val text = attributes.getInt(
             R.styleable.SignInWithAppleButton_sign_in_with_apple_button_textType,
             SignInTextType.SIGN_IN.ordinal
         )
 
+        // Corner radius
         val cornerRadius = attributes.getDimension(
             R.styleable.SignInWithAppleButton_sign_in_with_apple_button_cornerRadius,
```

Comment:
Would it make sense to define these as a default in the style too?

Reply:
In this case I think not, because per the HIG the coloring, text type, and corner radius should all be independent. I'm using styles for the coloring portion only.